### ISSUE TITLE
"error[call-non-callable]: Object of type `GenericAlias` is not callable" for instantiating Annoted classes

### ISSUE THREAD

**Issue by jankatins** [astral-sh/ty#3490]
### Summary

The following code works and mypy finds no issue with it. ty (and pyrefly) take issue:

```py
"""Test."""

from typing import Annotated
from fastapi import Depends

class _A:
    def __init__(self, txt: str):
        self.txt = txt

def get_a(txt: str) -> _A:
    return _A(txt)

A = Annotated[_A, Depends(get_a)]

a = A("test") # <- The problematic line
print(a)
```

```shell
$ mypy test.py             
Success: no issues found in 1 source file

$  python test.py              
<__main__._A object at 0x7f2aaf492660>

$ uvx ty check test.py     
error[call-non-callable]: Object of type `GenericAlias` is not callable
  --> test.py:15:5
   |
15 | a = A("test")
   |     ^^^^^^^^^
   |

Found 1 diagnostic
```
We have a lot of cases where we use this pattern to create fastapi dependencies and we want to use such dependencies in tests to create test data and now ty refuses this code.

### Version

ty 0.0.37

**Comment by charliermarsh** [comment]
Hmm, that seems like a clear violation of the [spec](https://typing.python.org/en/latest/spec/qualifiers.html#annotated)?

<img width="795" height="184" alt="Image" src="https://github.com/user-attachments/assets/d96e393d-a81d-4167-a732-5cfd491311ab" />

**Comment by AlexWaygood** [comment]
> Hmm, that seems like a clear violation of the [spec](https://typing.python.org/en/latest/spec/qualifiers.html#annotated)?

Yes -- previous discussion in https://github.com/astral-sh/ruff/pull/21672

**Comment by jankatins** [comment]
This is ... surprising, given that it works (at least our tests which do this a lot work) and mypy does not complain... Is this an error on mypy side? I found https://github.com/python/mypy/issues/10872 but not sure if this is the issue why mypy does not complain...

**Comment by AlexWaygood** [comment]
The typing spec was changed in https://github.com/python/typing/pull/1618 to add this restriction after a discussion in https://github.com/python/typing-council/issues/18. Mypy has been around for a long time and probably never changed its behaviour following that spec change.

As I said in https://github.com/astral-sh/ruff/pull/21672#issuecomment-3588472425, to me it makes sense to just emulate the runtime here. But with the spec the way it is, that would lead to us regressing our spec conformance score on the upstream tests.

**Comment by carljm** [comment]
I find the reasoning behind this spec change questionable, and would likely support an effort to change the spec here -- but it's also not currently enough of a priority for me to initiate or shepherd that change. If you have a strong use case for type checkers to support this, I'd suggest bringing it up on discuss.python.org, possibly followed by a PR to the typing spec and a typing-council decision issue.

Pending such a typing spec change, closing this as not planned.
